opm/simulators/wells/GasLiftSingleWell_impl.hpp

```diff
@@ -152,6 +149,36 @@ computeBhpAtThpLimit_(double alq) const
     return bhp_at_thp_limit;
 }
 
+template<typename TypeTag>
+void
+GasLiftSingleWell<TypeTag>::
+setupPhaseVariables_()
+{
+    const auto& pu = this->phase_usage_;
+    bool num_phases_ok = (pu.num_phases == 3);
+    if (pu.num_phases == 2
+        && pu.phase_used[BlackoilPhases::Aqua] == 1
+        && pu.phase_used[BlackoilPhases::Liquid] == 1
+        && pu.phase_used[BlackoilPhases::Vapour] == 0)
+    {
+        // NOTE: We support two-phase oil-water flow, by setting the gas flow rate
+        //   to zero. This is done by initializing the potential vector to zero:
+        //
+        //     std::vector<double> potentials(NUM_PHASES, 0.0);
+        //
+        //  see e.g. runOptimizeLoop_() in GasLiftSingleWellGeneric.cpp
+        //  In addition the VFP calculations, e.g. to calculate BHP from THP
+        //  has been adapted to the two-phase oil-water case, see the comment
+        //  in WellInterfaceGeneric.cpp for the method adaptRatesForVFP() for
+        //  more information.
+        num_phases_ok = true;  // two-phase oil-water is also supported
+    }
+    assert(num_phases_ok);
```

Comment:
I suggest a `throw` here that tells the user that we don't support oil-gas or water-gas